codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -482,6 +520,29 @@ class KotlinJmespathExpressionVisitor(
         return union
     }
 
+    private fun VisitedExpression.jmesPathToString(): String =
+        addTempVar("answer", "if(${this.identifier} as Any is String) ${this.identifier} else ${this.identifier}.toString()")
+
+    private fun VisitedExpression.toArray(): String =
+        addTempVar("answer", "if(${this.identifier} as Any is List<*> || ${this.identifier} as Any is Array<*>) ${this.identifier} as List<*> else listOf(${this.identifier})")
+
+    private fun List<VisitedExpression>.getNotNull(): String {
+        val answer = bestTempVarName("answer")
+        writer.write("var $answer: Any? = null")
+
+        val found = bestTempVarName("found")
+        writer.write("var $found = false")
+
+        forEach {
+            writer.withBlock("if (${it.identifier} != null && !$found) {", "}") {
+                write("$answer = ${it.identifier}")
+                write("$found = true")
+            }
+        }
+
+        return answer
+    }
```

Comment:
**Suggestion**: This should be possible without mutability and intermediate variables like `found`:

```kotlin
private fun List<VisitedExpression>.getNotNull(): String {
    val notNull = bestTempVarName("notNull")
    writer.withBlock("val $notNull = listOfNotNull(", ").firstOrNull()") {
        write("${it.identifier},")
    }

    return notNull
}
```

Which produces code like:

```kotlin
val notNull1 = listOfNotNull(
    foo,
    bar,
    baz,
).firstOrNull()
```